<launch>

<!-- NOTE: This requires DVO -- download and install separately! -->

  <node name="camera_tracker" pkg="dvo_ros" type="camera_tracker" respawn="false" output="screen">
  </node>

<param name="/camera_tracker/reconstruction" value="true"/>
<param name="/camera_tracker/use_weighting" value="true"/>
<param name="/camera_tracker/run_dense_tracking" value="true"/>
<param name="/camera_tracker/use_dense_tracking_estimate" value="true"/>

      <node name="omnimapper_ros_node" pkg="omnimapper_ros" type="omnimapper_openni" respawn="false" output="screen">
      	<param name="debug" value="true"/>
      	<param name="use_planes" value="false"/>
	    <param name="use_icp" value="true"/>
	    <param name="use_csm" value="false"/>
	    <param name="use_objects" value="false"/>
	    <param name="use_occ_edge_icp" value="false"/>
	    <param name="use_tf" value="true"/>
	    <param name="use_error_plugin" value="false"/>
	    <param name="use_tsdf_plugin" value="false"/>
	    <param name="use_no_motion" value="true"/>
	    <param name="use_organized_segmentation" value="false"/>
	    <param name="use_rgbd_sensor_base_tf_functor" value="false"/>
	    <param name="use_distortion_model" value="false"/>
      	<param name="odom_frame_name" value="/world"/>
	    <param name="base_frame_name" value="/base_link_estimate"/>
	    <param name="broadcast_map_to_odom" value="false"/>
	    <param name="broadcast_current_pose" value="true"/>
	    <param name="use_init_pose" value="false"/>
	    <param name="suppress_commit_window" value="true"/>
	    <param name="init_pose_from_tf" value="true"/>
	    <param name="cloud_topic_name" value="/camera/depth_registered/points"/>
	    <param name="icp_leaf_size" value="0.025"/>
	    <param name="icp_max_correspondence_distance" value="0.1"/>
	    <param name="icp_trans_noise" value="0.05"/>
	    <param name="icp_rot_noise" value="0.05"/>
	    <param name="icp_add_loop_closures" value="true"/>
	    <param name="occ_edge_trans_noise" value="0.05"/>
	    <param name="occ_edge_rot_noise" value="0.1"/>
	    <param name="occ_edge_max_correspondence_dist" value="0.05"/>
	    <param name="occ_edge_score_thresh" value="0.1"/>
	    <param name="occ_edge_add_identity_on_fail" value="false"/>
	    <param name="tf_trans_noise" value="10.05"/>
	    <param name="tf_rot_noise" value="0.0872664626"/>
	    <param name="tf_roll_noise" value="0.087"/>
	    <param name="tf_pitch_noise" value="0.087"/>
	    <param name="tf_yaw_noise" value="0.087"/>
	    <param name="draw_pose_array" value="true"/>
	    <param name="add_pose_per_cloud" value="true"/>
      </node>

       <node name="base_link_to_base_link_optical" pkg="tf" type="static_transform_publisher" args="0.0 0.0 0.0 1.57 -1.57 0.0 /current_pose /current_pose_optical 100" respawn="false" output="screen"/> 
	<!-- <node name="tf_world" pkg="tf" type="static_transform_publisher" args="0.465 -0.014 0.902 1.57 0 1.57 /odom /world 100" respawn="false" output="screen"/> -->
</launch>
